src/manager/TickManager.ts

```diff
@@ -1,46 +1,190 @@
+
+    if (leftToRight) {
+      const wordPos = compressed >> 8;
+      const minimum = (wordPos << 8) * tickSpacing;
+
+      if (this.isBelowSmallest(sortedTicks, tick)) {
+        return { nextTick: minimum, initialized: false };
+      }
+
+      const index = this.nextInitializedTick(sortedTicks, tick, leftToRight)
+        .tickIndex;
+      const nextInitializedTick = Math.max(minimum, index);
+      return {
+        nextTick: nextInitializedTick,
+        initialized: nextInitializedTick === index,
+      };
+    } else {
+      const wordPos = (compressed + 1) >> 8;
+      const maximum = ((wordPos + 1) << 8) * tickSpacing - 1;
+
+      if (this.isAtOrAboveLargest(sortedTicks, tick)) {
+        return { nextTick: maximum, initialized: false };
+      }
+
+      const index = this.nextInitializedTick(sortedTicks, tick, leftToRight)
+        .tickIndex;
+      const nextInitializedTick = Math.min(maximum, index);
+      return {
+        nextTick: nextInitializedTick,
+        initialized: nextInitializedTick === index,
+      };
+    }
   }
 
   getFeeGrowthInside(
     tickLower: number,
     tickUpper: number,
     tickCurrent: number,
     feeGrowthGlobal0X128: JSBI,
-    feeGrowthGlobal1X128: JSBI,
-    leftToRight: boolean
-  ): JSBI {
-    // TODO
-    return JSBI.BigInt(0);
+    feeGrowthGlobal1X128: JSBI
+  ): { feeGrowthInside0X128: JSBI; feeGrowthInside1X128: JSBI } {
+    const lower = this.get(tickLower);
+    const upper = this.get(tickUpper);
+    let feeGrowthBelow0X128: JSBI;
+    let feeGrowthBelow1X128: JSBI;
+    if (tickCurrent >= tickLower) {
+      feeGrowthBelow0X128 = lower.feeGrowthOutside0X128;
+      feeGrowthBelow1X128 = lower.feeGrowthOutside1X128;
+    } else {
+      feeGrowthBelow0X128 = JSBI.subtract(
+        feeGrowthGlobal0X128,
+        lower.feeGrowthOutside0X128
+      );
+      feeGrowthBelow1X128 = JSBI.subtract(
+        feeGrowthGlobal1X128,
+        lower.feeGrowthOutside1X128
+      );
+    }
+
+    let feeGrowthAbove0X128: JSBI;
+    let feeGrowthAbove1X128: JSBI;
+    if (tickCurrent < tickUpper) {
+      feeGrowthAbove0X128 = upper.feeGrowthOutside0X128;
+      feeGrowthAbove1X128 = upper.feeGrowthOutside1X128;
+    } else {
+      feeGrowthAbove0X128 = JSBI.subtract(
+        feeGrowthGlobal0X128,
+        upper.feeGrowthOutside0X128
+      );
+      feeGrowthAbove1X128 = JSBI.subtract(
+        feeGrowthGlobal1X128,
+        upper.feeGrowthOutside1X128
+      );
+    }
+    return {
+      feeGrowthInside0X128: JSBI.subtract(
+        JSBI.subtract(feeGrowthGlobal0X128, feeGrowthBelow0X128),
+        feeGrowthAbove0X128
+      ),
+      feeGrowthInside1X128: JSBI.subtract(
+        JSBI.subtract(feeGrowthGlobal1X128, feeGrowthBelow1X128),
+        feeGrowthAbove1X128
+      ),
+    };
+  }
+
+  clear(tick: number) {
+    this.ticks.delete(tick);
+  }
+
+  private getSortedTicks(): Tick[] {
+    const sortedTicks = new Map([...this.ticks.entries()].sort());
```

Comment:
We need to add a comparator then the `sort` method will act as we expected.
/**
     * Sorts an array in place.
     * This method mutates the array and returns a reference to the same array.
     * @param compareFn Function used to determine the order of the elements. It is expected to return
     * a negative value if first argument is less than second argument, zero if they're equal and a positive
     * value otherwise. If omitted, the elements are sorted in ascending, ASCII character order.
     * ```ts
     * [11,2,22,1].sort((a, b) => a - b)
     * ```
     */
    sort(compareFn?: (a: T, b: T) => number): this;

Reply:
![image](https://user-images.githubusercontent.com/60888366/132191316-9b147232-42f9-4aba-8a49-1184f1fb76ea.png)

Reply:
I believe `map.entries()` returns the key-value pair and it sorts in ascending order by keys, in our case, `number`, I doubt it would require an explicit comparator.

Reply:
I see your point. Indeed, this sort is doing by ASCII character order of keys. However, guess we both experienced sorting problem on column of string / varchar type on some db, I think they are of the same issue.

Reply:
I stand corrected, the importance of test cases is real.

Reply:
We exactyl need a test for this type of case, very easy to confused. Cool!